### DISCUSSION TITLE
Autoortho Kills Internet Connection

### DISCUSSION THREAD

**Discussion by within7** [kubilus1/autoortho#285] [General]
Hello,

When I launch Autoortho and hit "Start new flight" in XP11, when the Autoortho CMD starts writing down with lot of lines, suddenly the internet connection is being lost - ALL devices connected to my Wi-Fi lose the connection. This gets resolved when I shutdown Autoortho.

What this might be related and how can I fix it?

**Comment by kubilus1** [comment]
It's probably related to your DNS setup most likely, or perhaps you have a firewall that is blocking when there are too many outbound connections.  You'll need to look into resolving your network issue since this project will by it's nature make many outbound connections.

**Reply by within7** [reply]
I have recently formatted my PC so I didn't touch anything related to DNS Setup. Is there a DNS Setup that you can recommend? I believe I was using 8.8.8.8/8.8.4.4 before I formatted my PC and it worked fine with Autoortho. I tried that DNS configuration after formatting but it didn't work :/

Is there anything that you can recommend?

**Reply by elvenbone** [reply]
Can confirm, AuthoOrtho makes almost 1000 DNS requests per second (on my desktop machine(TM)). Loading a flight in a blank location (e.g. on first use of AuthoOrtho) with min. ZL16 produces ~1.3 million DNS requests in total.

On my local pi-hole DNS resolver rate-limiting kicked in, effectively killing the internet connection of the PC. Had to disable it, but maybe the ISP or public resolver used by @within7 applies a similar rate-limiting.

It is very strange that AuthoOrtho does a look-up each time instead of caching the IP addresses. After all it is only the same four domains:

| domain                   | number of queries |
| r0.ortho.tiles.virtualearth.net | 335353 |
| r1.ortho.tiles.virtualearth.net | 335205 |
| r2.ortho.tiles.virtualearth.net | 332378 |
| r3.ortho.tiles.virtualearth.net | 320070 |

**Comment by kubilus1** [comment]
So on Linux there is this thing called dnsmasq that does that for you.
It's kind of weird and janky that Windows doesn't do something similar.

**Reply by elvenbone** [reply]
The PC is Arch Linux. Dnsmasq is what the pi-hole uses for the whole home network. Having such a DNS resolver installed locally on the PC would be quite untypical, even more on Windows systems.

**Comment by kubilus1** [comment]
So local DNS cache is pretty typical.  On modern Ubuntu this is via
systemd-resolved
https://manpages.ubuntu.com/manpages/impish/man8/systemd-resolved.service.8.html.
It's been a bit but prior to that I believe this was via dnsmasq.  On Arch,
I can't say, but you may want to configure local cache as that may speed a
lot of things up for you.
